boa3_test/tests/examples_tests/test_htlc.py

```diff
@@ -1,477 +1,324 @@
+                                    target_contract=constants.NEO_SCRIPT)
+        self.assertEqual(True, result)
+        result, _ = await self.call('transfer',
+                                    [person_b, htlc, transferred_amount_gas, None],
+                                    signing_accounts=[self.account2],
+                                    return_type=bool,
+                                    target_contract=constants.GAS_SCRIPT)
+        self.assertEqual(True, result)
+
+        await asyncio.sleep(self.REFUND_WAIT_TIME)
         # will be able to refund, because enough time has passed
-        invoke_refund_success = runner.run_contract(path, 'refund', account=self.OWNER)
-
-        runner.update_contracts(export_checkpoint=True)
-
-        # The `refund` method uses runtime.time, but BoaTestRunner fails to get it
-        transaction_refund = self._validate_execution_result(runner, invoke_refund_success.tx_id, expected_result=True)
+        invoke_refund_success, notifications = await self.call('refund', [], return_type=bool, signing_accounts=[self.owner])
+        self.assertEqual(True, invoke_refund_success)
 
         # person_a and person_b transferred cryptocurrency to the contract, so they both will be refunded
-        transfer_events = []
-        # removing possible GAS minting from the List
-        for k in transaction_refund.notifications:
-            if k.arguments[0] is not None and k.name == 'Transfer':
-                transfer_events.append(k)
-        self.assertEqual(2, len(transfer_events))
-        self.assertEqual(3, len(transfer_events[0].arguments))
-        self.assertEqual(3, len(transfer_events[1].arguments))
+        transfer_events = self.filter_events(
+            notifications,
+            event_name='Transfer',
+            notification_type=boatestcase.Nep17TransferEvent
+        )
+        filtered_transfer_events = list(filter(lambda x: x.source is not None, transfer_events))
+        self.assertEqual(2, len(filtered_transfer_events))
 
         # HTLC returning the tokens
-        sender, receiver, amount = transfer_events[0].arguments
-        self.assertEqual(htlc_script_hash, sender)
-        self.assertEqual(person_a_script_hash, receiver)
-        self.assertEqual(transferred_amount_neo, amount)
-
-        sender, receiver, amount = transfer_events[1].arguments
-        self.assertEqual(htlc_script_hash, sender)
-        self.assertEqual(person_b_script_hash, receiver)
-        self.assertEqual(transferred_amount_gas, amount)
+        self.assertEqual(htlc, filtered_transfer_events[0].source)
+        self.assertEqual(person_a, filtered_transfer_events[0].destination)
+        self.assertEqual(transferred_amount_neo, filtered_transfer_events[0].amount)
+
+        self.assertEqual(htlc, filtered_transfer_events[1].source)
+        self.assertEqual(person_b, filtered_transfer_events[1].destination)
+        self.assertEqual(transferred_amount_gas, filtered_transfer_events[1].amount)
```

Comment:
when there's more than one event expected, you should use variables to  let the test more readable
```suggestion
        refund_A, refund_B = filtered_transfer_events
        
        self.assertEqual(htlc, refund_A.source)
        self.assertEqual(person_a, refund_A.destination)
        self.assertEqual(transferred_amount_neo, refund_A.amount)

        self.assertEqual(htlc, refund_B.source)
        self.assertEqual(person_b, refund_B.destination)
        self.assertEqual(transferred_amount_gas, refund_B.amount)
```